### PR TITLE
Add test to ensure correct order of repair queue selection

### PR DESCRIPTION
**Pull request by mobyvb**
What: 
Add satellitedb test to make sure we remove from the repair queue in the following order:
1. attempted = null
2. attempted in ascending order (older timestamps first)

Why:
We recently merged a change (#2533) to update the order used for returning items in the repair queue, but the test didn't cover some details

Please describe the tests:
 - Test 1:
Add four paths to the repair queue (nullPath, recentRepairPath, oldRepairPath, and olderRepairPath)
Update recentRepairPath to have a last attempted time of now
Update oldRepairPath to have a last attempted time of 2 hours ago
Update olderRepairPath to have a last attempted time of 3 hours ago
Select from the repair queue 4 times. Expect nullPath first, olderRepairPath second, oldRepairPath third, and an empty queue error the fourth time.
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/repairqueue_test.go [hunk 1]
```diff
@@ -48,3 +53,78 @@ func TestRepairQueue(t *testing.T) {
 		}
 	})
 }
+
+func TestRepairQueueOrder(t *testing.T) {
```
Comment (ifraixedes) [inline satellite/satellitedb/repairqueue_test.go]: I think that the description of the test that you wrote in the issue could be added here.

File: satellite/satellitedb/repairqueue_test.go [hunk 2]
```diff
@@ -48,3 +53,78 @@ func TestRepairQueue(t *testing.T) {
...
+			injuredSeg := &pb.InjuredSegment{Path: path}
+			err := repairQueue.Insert(ctx, injuredSeg)
+			require.NoError(t, err)
+		}
+
+		dbAccess := db.(interface{ TestDBAccess() *dbx.DB }).TestDBAccess()
+		var timeConvertPrefix string
+		switch d := dbAccess.DB.Driver().(type) {
+		case *sqlite3.SQLiteDriver:
+			timeConvertPrefix = "datetime("
```
Comment (ifraixedes) [inline satellite/satellitedb/repairqueue_test.go]: I've seen in some PRs that we're moving away from using `datetime` in sqlite due the precision that it has. Shouldn't this test be updated accordingly?

Comment (mobyvb) [reply satellite/satellitedb/repairqueue_test.go]: I was not aware that we were moving away from datetime, but the repair queue select query itself uses datetime, so I wanted to be consistent here

Comment (aligeti) [reply satellite/satellitedb/repairqueue_test.go]: Moby the date time is UTC based? make sure the timing is handled correctly. Other than that test does what is explained in the PR.

Comment (mobyvb) [reply satellite/satellitedb/repairqueue_test.go]: yeah, it should be utc based after I did some searching. These are also the same strings we use for postgres vs. sqlite elsewhere in the codebase

**Pull request merged**